### DISCUSSION TITLE
Social Network Analysis

### DISCUSSION THREAD

**Discussion by jarrodmillman** [networkx/networkx#4444] [Algorithms]
What algorithms should we add?

**Comment by dschult** [comment]
Louvain Community Detection algorithms.
I have some work done by 2 student honors theses that I hope to whip into shape in Jan 2021, but there are many such approaches and we will likely want/need more than one approach.

**Reply by BradKML** [reply]
This will help the issue. https://github.com/GiulioRossetti/cdlib
But imagine something more advanced like https://github.com/benedekrozemberczki/karateclub

**Comment by MridulS** [comment]
There is also Leiden (improvement over Louvain). https://github.com/vtraag/leidenalg works natively with igraph, and they don't plan to support NetworkX any time soon. (https://github.com/vtraag/leidenalg/issues/54).
Does it make sense to have an example of this in the documentation? (Conversion to igraph -> call the algorithm -> back to networkx)

**Reply by dschult** [reply]
Nice!    
In the readme for that package it says that igraph now provides Leiden natively, "probably much faster"  though with less flexibility.  So if we do an example we wouldn't have to install leidenalg... just igraph.

**Reply by BradKML** [reply]
Check this bruv https://github.com/GiulioRossetti/cdlib

**Comment by BradKML** [comment]
There are some paper digging I did for this question: 
Community Detection https://github.com/networkx/networkx/discussions/4809
Centrality and Validation https://github.com/networkx/networkx/discussions/4674
Role Detection or Role Discovery https://github.com/networkx/networkx/discussions/4673
